webapp/src/Controller/API/ContestController.php

```diff
@@ -200,6 +208,90 @@ public function bannerAction(Request $request, string $id): Response
         return static::sendBinaryFileResponse($request, $banner, 'image/png');
     }
 
+    /**
+     * Delete the banner for the given contest
+     * @Rest\Delete("/{id}/banner.png", name="delete_contest_banner")
+     * @IsGranted("ROLE_ADMIN")
+     * @OA\Response(response="204", description="Deleting banner succeeded")
+     * @OA\Parameter(ref="#/components/parameters/id")
+     */
+    public function deleteBannerAction(Request $request, string $id): Response
+    {
+        /** @var Contest $contest */
+        $contest = $this->getQueryBuilder($request)
+            ->andWhere(sprintf('%s = :id', $this->getIdField()))
+            ->setParameter(':id', $id)
+            ->getQuery()
+            ->getOneOrNullResult();
+
+        if ($contest === null) {
+            throw new NotFoundHttpException(sprintf('Object with ID \'%s\' not found', $id));
+        }
+
+        $contest->setClearBanner(true);
+
+        $this->assetUpdater->updateAssets($contest);
+        $this->eventLogService->log('contests', $contest->getCid(), EventLogService::ACTION_UPDATE,
+            $contest->getCid());
+
+        return new Response('', Response::HTTP_NO_CONTENT);
+    }
+
+    /**
+     * Set the banner for the given contest
+     * @Rest\POST("/{id}/banner.png", name="post_contest_banner")
+     * @Rest\PUT("/{id}/banner.png", name="put_contest_banner")
+     * @OA\RequestBody(
+     *     required=true,
+     *     @OA\MediaType(
+     *         mediaType="multipart/form-data",
+     *         @OA\Schema(
+     *             required={"banner"},
+     *             @OA\Property(
+     *                 property="banner",
+     *                 type="string",
+     *                 format="binary",
+     *                 description="The banner to use."
+     *             )
+     *         )
+     *     )
+     * )
+     * @IsGranted("ROLE_ADMIN")
+     * @OA\Response(response="400", description="Invalid data provided")
+     * @OA\Response(response="204", description="Setting banner succeeded")
+     * @OA\Parameter(ref="#/components/parameters/id")
+     */
+    public function setBannerAction(Request $request, string $id): Response
+    {
+        /** @var Contest $contest */
+        $contest = $this->getQueryBuilder($request)
+            ->andWhere(sprintf('%s = :id', $this->getIdField()))
+            ->setParameter(':id', $id)
+            ->getQuery()
+            ->getOneOrNullResult();
+
+        if ($contest === null) {
+            throw new NotFoundHttpException(sprintf('Object with ID \'%s\' not found', $id));
+        }
+
+        /** @var UploadedFile $banner */
+        $banner = $request->files->get('banner');
+
+        if (!$banner) {
+            throw new BadRequestHttpException("Please supply a banner");
+        } else if ($banner->getMimeType() !== 'image/png') {
```

Comment:
Have we always only supported PNG? Why not also support JPEG and maybe SVG too (although I understand that might be more tricky with pixel size)?

Reply:
We have indeed only supported PNG for these (and JPG for photo's). I'd like to add support for all three, but I want to do that in a separate PR since it involves more changes (i.e. also the GET request in the API).

Reply:
> do we want to / need to make sure that the assets are not too large?

If we add SVG support (which I would like) this is already becoming hard. Or are you proposing filesize? We can add that. What do you propose in that case for size?

Reply:
svg is also dangerous so likely we'd need some validation?

Reply:
Only admins can do all this right? I mean, you can also upload malicious problems in that case I guess?